search_engine_parser/core/engines/google.py

```diff
@@ -32,6 +32,71 @@ def parse_soup(self, soup):
         # find all class_='g' => each result
         return soup.find_all('div', class_='g')
 
+    def parse_result(self, results, **kwargs):
+        """
+        Runs every entry on the page through parse_single_result
+
+        :param results: Result of main search to extract individual results
+        :type results: list[`bs4.element.ResultSet`]
+        :returns: dictionary. Containing lists of titles, links, descriptions, direct results and other possible\
+            returns.
+        :rtype: dict
+        """
+        search_results = dict()
+        for each in results:
+            try:
+                rdict = self.parse_single_result(each, **kwargs)
+                # Create a list for all keys in rdict if not exist, else
+                for key in rdict.keys():
+                    if key not in search_results.keys():
+                        search_results[key] = list([rdict[key]])
+                    else:
+                        search_results[key].append(rdict[key])
+            except Exception: #pylint: disable=invalid-name, broad-except
+                pass
+        search_results['direct_answer'] = self.parse_direct_answer(results[0])
```

Comment:
Can you make this 
```python
direct_answer = self.parse_direct_answer(results[0])
if direct_answer:
	search_results['direct_answer'] = direct_answer
```

just so there's no None values for results that don't have direct Answers

Reply:
This is already implemented in parse_direct_answer
```
def parse_direct_answer(self, single_result):
        direct_answer = ''
```
direct_answer is initialized as an empty string. It will return an empty string if the results don't have direct Answers.

Reply:
I can add this code if needed
```
direct_answer = self.parse_direct_answer(results[0])
if direct_answer:
	search_results['direct_answer'] = direct_answer
```

Reply:
Done the suggestion implemented